armi/physics/neutronics/globalFlux/globalFluxInterface.py

```diff
@@ -1169,3 +1169,11 @@ def calcReactionRates(obj, keff, lib):
 
     for paramName, val in rate.items():
         obj.p[paramName] = val  # put in #/cm^3/s
+
+    if rate["rateFis"] > 0.0:
+        vFuel = obj.getComponentAreaFrac(Flags.FUEL)
+        obj.p.fisDens = rate["rateFis"] / vFuel
+        obj.p.fisDensHom = rate["rateFis"]
+    else:
```

Comment:
Sorry, naïve question.

Is this because `"rateFis"` might be zero?  Or are you trying to catch some crazy "negative rate" error?

Reply:
The goal was to not call `obj.getComponentAreaFrac(Flags.FUEL)` on a block unless it has a positive fission rate. If we call that on every block, then we'll get a bunch of warnings from this: https://github.com/terrapower/armi/blob/76a76b1b404a7beb38282c16405c25b9d54fe57a/armi/reactor/blocks.py#L1092-L1098